test/runtests.jl

```diff
@@ -358,6 +358,31 @@ end
     @test_throws DomainError interpolants(r, [2.5, NaN])
 end
 
+@testset "Iteration and Indexing" begin
+    @testset "2d" begin
+        g = GridInterpolations.RectangleGrid(1:3,4:5)
+        pts = ([1.,4.],[2.,4.],[3.,4.],[1.,5.],[2.,5.],[3.,5.])
+        @test length(g) == length(pts)
+        @test all(pts .== [x for x in g])
+        @test iterate(g,7) === nothing
+        @test all(pts .== [g[ci] for ci in reshape(CartesianIndices((3,2)), length(g))])
```

Comment:
Is the `CartesianIndices` constructor needed or can one just index using `g[3, 2]` for example?

Reply:
Support for multi-indices added in latest commit (although Cartesian coordinates seem to be safer generally)